Data_Structures_Answers.md

```diff
@@ -1,18 +1,25 @@
 Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+O(1)
 
 2. What is the space complexity of your ring buffer's `append` function?
+O(1)
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+O(n)
 
 4. What is the space complexity of your ring buffer's `get` method?
-
+O(n)
 
 5. What is the runtime complexity of the provided code in `names.py`?
+O(n^2)
 
 6. What is the space complexity of the provided code in `names.py`?
+O(n^2)
 
 7. What is the runtime complexity of your optimized code in `names.py`?
+O(n)
```

Comment:
This is going to be O(nlog(n)) since we used the Binary search tree for the lookup table.  Since a BST get method has log(n) time complexity and we do this n times, we have nlog(n) complexity.